### PR TITLE
SVCs to output actual probabilities

### LINKED ISSUE #192: Possible bug: 'LinearSVC' object has no attribute 'predict_proba'
**Bug Description**
LinearSVC [evidently does not](https://scikit-learn.org/stable/modules/svm.html#scores-and-probabilities) generate class membership probability estimates so `classify-samples` fails with that estimator (see [forum xref](https://forum.qiime2.org/t/sample-classifier-error-shape-of-passed-values-is-73-1-indices-imply-73-2/14204?u=nicholas_bokulich))

However, it is evidently passing in the unit tests:
https://github.com/qiime2/q2-sample-classifier/blob/5a9b78d192d4897fd41c3648180bf3f8898d46e7/q2_sample_classifier/tests/test_estimators.py#L233-L252

**Expected behavior**
This should pass; the sklearn docs recommend setting `LinearSVC(probability=True)` so perhaps this is a version issue and this should be set explicitly here:
https://github.com/qiime2/q2-sample-classifier/blob/5a9b78d192d4897fd41c3648180bf3f8898d46e7/q2_sample_classifier/utilities.py#L767

More investigation is needed to determine:
1. Why the unit tests are passing with LinearSVC
2. Whether LinearSVC passes on real datasets (and if so, why that forum user's tests are failing, perhaps an sklearn version issue?)

### PR DESCRIPTION
**Pull request by adamovanja**
closes #192

`LinearSVC(probability=True)` is no longer supported in the current sklearn version ([source](https://scikit-learn.org/stable/modules/generated/sklearn.svm.LinearSVC.html)). Also, the previous implementations of `SVC_linear` & `SVC` return (when used with the newest sklearn version) the classifiers' decision functions not the predicted probability values ([see decision_function description](https://scikit-learn.org/stable/modules/generated/sklearn.svm.SVC.html#sklearn.svm.SVC.decision_function))

Hence, I suggest to make `SVC` and `SVC_linear` return calibrated probability values as described [here](https://scikit-learn.org/stable/modules/svm.html#scores-probabilities) and to adjust the `SVC_linear` classifier to be defined by `SVC(kernel='linear')` instead of `LinearSVC()`. This PR further includes a verification for all classifiers to output predicted probabilities in a range from 0 to 1.

### CODE REVIEW

**Comment by thermokarst** [comment]
Awesome, thanks for this @adamovanja! I put a "do not merge" on this PR because I am getting ready to upgrade this plugin to python 3.8 (later today) - once I do that we can remove the label and re-run the unit tests on python 3.8 to make sure that everything still works as expected. Thanks!

**Comment by adamovanja** [comment]
hi @thermokarst, your python 3.8 changes are now merged and all tests run smoothly. So, I think everything is set to remove the  "do not merge" label.

**Review by nbokulich — APPROVED** [review]
Looks good to me @adamovanja ! Ready to merge once @thermokarst gives the signal (looks like the python upgrade PR was merged, just want to make sure we are not waiting on anything else)

**Comment by thermokarst** [comment]
Thanks @adamovanja & @nbokulich - I'm not quite done with the Python 3.8 migration, but hopefully later today - I'll merge this once the pipeline is cleared! Thanks!

**Review by thermokarst — APPROVED** [review]
